backend/src/test/java/de/ffmjava/capstone/backend/horses/HorseServiceTest.java

```diff
@@ -23,4 +25,43 @@ void getAllHorses() {
 
         assertEquals(expected, actual);
     }
-}
\ No newline at end of file
+
+    @Test
+    void addNewHorse_AndExpectHorse() {
+        Horse newHorse = new Horse(null, "name", "owner", null);
+
+        doReturn(newHorse.withId("1")).when(mockRepository).save(any());
```

Comment:
is there a difference to our usual when(repo.method()).thenReturn() ?

and you don't test the correct adding of an ID...

Reply:
It does the same, can be also used with static methods though